views/components/config_card.erb

```diff
@@ -0,0 +1,95 @@
+            <input
+              type="text"
+              class="border rounded py-1 focus:ring-blue-500 focus:border-blue-500 mr-2 <%= (errors[key] && !errors[key].is_a?(Array)) ? "border-red-500" : "" %>"
+              value="<%= key %>"
+              name="<%= name %>_keys[]"
+              required
+            />
+            <input
+              type="text"
+              value="<%= value %>"
+              class="border rounded py-1 focus:ring-blue-500 focus:border-blue-500 <%= errors[key].is_a?(Array) ? "border-red-500" : "" %>"
+              name="<%= name %>_values[]"
+            />
+            <% if errors[key] %>
+              <span class="error inline-block h-8 w-54 py-2 px-2 max-w-54 text-xs text-red-600 bg-red-100 rounded" aria-live="polite">
+                <% if errors[key].is_a?(Array) %>
+                  <% errors[key].each do |error| %>
+                    <%= error %><br/>
+                  <% end %>
+                <% else %>
+                  <%= errors[key] %>
+                <% end %>
+              </span>
+            <% end %>
+          </div>
+          <div class="min-w-16 flex items-center ml-2">
+            <%== part("components/button", type: "danger", text: "", icon: "hero-trash", extra_class: "delete-config-btn") if @edit_perm %>
+          </div>
+        </div>
+      </div>
+    <% end %>
+    <!-- Hidden row for generating new entries -->
+    <div class="px-6 py-2 hover:bg-gray-50 group config-placeholder-group font-mono hidden" data-config-id="">
+      <div class="flex items-center justify-between">
+        <div class="flex flex-wrap items-center config-entry">
+          <input
+            type="text"
+            class="border rounded py-1 focus:ring-blue-500 focus:border-blue-500 mr-2"
+            value=""
+            name="<%= name %>_keys[]"
+            disabled
+          />
+          <input
+            type="text"
+            value=""
+            class="border rounded py-1 focus:ring-blue-500 focus:border-blue-500"
+            name="<%= name %>_values[]"
+            disabled
+          />
+          <span class="error inline-block h-8 w-54 py-2 px-2 max-w-54 text-xs text-red-600 hidden bg-red-100 rounded" aria-live="polite"></span>
+        </div>
+        <div class="min-w-16 flex items-center ml-2">
+          <%== part("components/button", type: "danger", text: "", icon: "hero-trash", extra_class: "delete-config-btn") %>
+        </div>
+      </div>
+    </div>
+    <!-- Empty row for new configuration -->
+    <% if @edit_perm %>
+      <div class="px-6 py-2 hover:bg-gray-50 group new-config font-mono">
+        <div class="flex items-center justify-between">
+          <div class="flex flex-wrap items-center">
+            <input
+              type="text"
+              class="border rounded py-1 focus:ring-blue-500 focus:border-blue-500 mr-2"
+              placeholder="New Key"
+              name="keys_placeholder"
```

Comment:
For the empty row, I would use`<%= name %>_keys[]` and  `<%= name %>_values[]` as the input names. This will ensure the empty row is submitted with a normal form submission. This should allow you to eliminate the javascript handler on on the save button.

Reply:
Sounds good, skipping the row makes it work fine, even with `required` on the existing config rows. I think we should leave it this way to avoid empty key submits.